staging/src/k8s.io/cluster-bootstrap/util/tokens/tokens.go

```diff
@@ -14,16 +14,21 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1beta2
+package tokens
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 
 	"github.com/pkg/errors"
 
-	bootstrapapi "k8s.io/cluster-bootstrap/token/api"
-	bootstraputil "k8s.io/cluster-bootstrap/token/util"
+	"k8s.io/cluster-bootstrap/token/api"
+	legacyutil "k8s.io/cluster-bootstrap/token/util"
+)
+
+var (
+	bootstrapTokenRegexp = regexp.MustCompile(api.BootstrapTokenPattern)
 )
 
 // BootstrapTokenString is a token of the format abcdef.abcdef0123456789 that is used
```

Comment:
This has always looked to me as a versioned API and, indeed, it is part of the kubeadm versioned APIs. Perhaps we should introduce such an API as part of cluster-bootstrap?

@dixudx @neolit123 @fabriziopandini WDYT?

Reply:
yes, and we need to poke api-machinery about:
https://github.com/kubernetes/cluster-bootstrap#things-you-should-not-do
> Add API types to this repo. This is for the helpers, not for the types.

Reply:
the alternative is to not have BoostrapTokenString as a public type under cluster-bootstrap, but rather as a private type with public helpers.

Reply:
it seems like https://github.com/kubernetes/cluster-bootstrap/blob/master/token/api landed last month (even though it contains only constants for now

Reply:
the only API is the type, structure, and helper methods to parse the bootstrap token secret contents, not the structures used to store decomposed bootstrap tokens in kubeadm config files